component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -182,9 +187,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                         }
                     }
                 }
-            } else if (savedOTPString == null) {
-                throw new AuthenticationFailedException("The claim " + SMSOTPConstants.SAVED_OTP_LIST +
-                        " does not contain any values");
+            } else if (codeBool == "true") {
```

Comment:
Also why can't we add this condition to following.
https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/master/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java#L154